google/cloud/bigtable/row_data.py

```diff
@@ -400,7 +400,7 @@ def __init__(self, read_method, request, retry=DEFAULT_RETRY_READ_ROWS):
         self.read_method = read_method
         self.request = request
         self.retry = retry
-        self.response_iterator = read_method(request)
+        self.response_iterator = read_method(request, timeout=self.retry._deadline + 1)
```

Comment:
is there a test we can add here? (or are we waiting on the release in python-api-core?)

Reply:
@kolea2 The injected `retry.timeout` parameter propagates implicitly to `grpcio.experimental.aio._channel.UnaryStreamMultiCallable` class, hence the best way to test it within the scope of CBT library is via checking the contents of `self.retry` member of the `row_data.PartialRowData` class. I've added such a line to an existing test method along with a couple of other fixes for tests that failed upon introducing the proposed changes.